docs/js/instantsearch.js

```diff
@@ -0,0 +1,170 @@
+    const parsed_search_query = decodeURI(search_query.replace('+', ' '));
+    const search_page = doc.location.hash.match(/p=(\d*?)(&|$)/)[1] ?? 1;
+    const parsed_search_page = parseInt(search_page);
+    const version = doc.location.pathname.split('/')[2];
+    const search = instantsearch({
+        indexName: 'ezplatform',
+        searchClient: algoliasearch('2DNYOU6YJZ', '21ce3e522455e18e7ee16cf7d66edb4b'),
+        initialUiState: {
+            ezplatform: {
+                query: parsed_search_query,
+                refinementList: {version: [version]},
+                page: parsed_search_page,
+            },
+        },
+    });
+
+    doc.getElementById('searchbox').addEventListener('keyup', function (event) {
+        window.location.hash = '#q=' + encodeURI(event.target.value) + '&p=1';
+    })
+
+    doc.getElementById('pagination').addEventListener('click', function (event) {
+        let page = doc.getElementsByClassName('ais-Pagination-item--selected').length ? parseInt(doc.getElementsByClassName('ais-Pagination-item--selected')[0].innerText) : 1
+        window.location.hash = window.location.hash.includes('p=') ? window.location.hash.replace(/p=\d*/, 'p=' + page) : window.location.hash + '&p=' + page;
+    })
+
+    search.addWidgets([
+        instantsearch.widgets.configure({
+            hitsPerPage: 10,
+        }),
+        instantsearch.widgets.stats({
+            container: '#stats',
+            templates: {
+            text: `<h1>
+                Search results ({{#helpers.formatNumber}}{{nbHits}}{{/helpers.formatNumber}})
+            </h1>`,
+            },
+        }),
+        instantsearch.widgets.searchBox({
+            container: '#searchbox',
+        }),
+        instantsearch.widgets.hits({
+            container: '#hits',
+            transformItems(items) {
+                const outputItemsMap = {};
+
+                items.forEach((item) => {
+                    const hierarchy = Object.entries(item.hierarchy).filter(([, value]) => value);
+                    const breadcrumbsKeys = hierarchy.map(([key]) => key);
+                    const groupNameKey = breadcrumbsKeys.shift();
+                    const entryNameKey = breadcrumbsKeys.shift();
+                    const newItem = {
+                        entryNameKey,
+                        breadcrumbsKeys,
+                        item: getClearedItem(item),
+                    };
+                    const groupChildren = outputItemsMap[groupNameKey]?.children ?? [];
+
+                    outputItemsMap[groupNameKey] = {
+                        groupNameKey,
+                        children: [...groupChildren, newItem],
+                    };
+                });
+
+                return Object.values(outputItemsMap);
+            },
+            templates: {
+                item: (hit) => {
+                    const { groupNameKey, children } = hit;
+                    const groupItem = children[0].item;
```

Comment:
This is wrong, not all hits on a page are in the same group.

Reply:
Fixed in 91f98906b36d5f5f1976ea95c8842f7648b935de where there is room for refactoring.